projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -1095,22 +1095,10 @@ export class DatatableComponent<TRow extends Row = any>
       return;
     }
 
-    let idx: number;
-    const cols = this._internalColumns.map((c, i) => {
-      c = { ...c };
-
-      if (c.$$id === column.$$id) {
-        idx = i;
-        c.width = newValue;
-
-        // set this so we can force the column
-        // width distribution to be to this value
-        c.$$oldWidth = newValue;
-      }
-
-      return c;
-    });
-
+    const idx = this._internalColumns.indexOf(column);
```

Comment:
What's the reason to search for the full column here (that could in the worst case change) and not like the logic before, just search for the index?

```suggestion
    const idx = this._internalColumns.findIndex(c => c.$$id === column.$$id);
```

Reply:
As offline discussed:

Please check a few lines below, the other similar handler is already checking by ref.
I think it was previously just written that weird because the first changed the reference and then try to find the column.
I could not find any other relevant location where the column refs are changed which could lead to issues.

And it works ...

Reply:
@spike-rabbit thanks for the feedback. As also discussed: Migrating completely to reference handling makes, we should just avoid handling matching both in a mix of references and ID. If you keep track of that and we have a clear migration strategy to handle this everwhere the same, I'm perfectly fine with it.